src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -1248,12 +1260,7 @@ - (NSUInteger)unitTestNonnullDelegateCount
                                                             attributeID:(NSNumber *)attributeID
                                                                  params:(MTRReadParams * _Nullable)params
 {
-#define MTRDeviceErrorStr "MTRDevice readAttributeWithEndpointID:clusterID:attributeID:params: must be handled by subclasses"
-    MTR_LOG_ERROR(MTRDeviceErrorStr);
-#ifdef DEBUG
-    NSAssert(NO, @MTRDeviceErrorStr);
-#endif // DEBUG
-#undef MTRDeviceErrorStr
+    MTR_REQUIRE_SUBCLASS_IMPL("readAttributeWithEndpointID:clusterID:attributeID:params:");
```

Comment:
if above suggestion for variadic macros is accepted, all following `MTR_REQUIRE_SUBCLASS_IMPL(...)`s can just be
```suggestion
    MTR_REQUIRE_SUBCLASS_IMPL();
```